pow/src/test/java/tech/pegasys/teku/pow/Web3jEth1MonitorableProviderTest.java

```diff
@@ -0,0 +1,244 @@
+
+    succeedAValidation();
+
+    // just validated, no need to validate
+    assertThat(provider.needsToBeValidated()).isFalse();
+
+    timeProvider.advanceTimeBySeconds(
+        Constants.ETH1_VALID_ENDPOINT_CHECK_INTERVAL.getSeconds() + 1);
+
+    // after the interval needs to be validated
+    assertThat(provider.needsToBeValidated()).isTrue();
+
+    succeedAValidation();
+
+    failACall();
+
+    // advance less then required
+    timeProvider.advanceTimeBySeconds(
+        Constants.ETH1_FAILED_ENDPOINT_CHECK_INTERVAL.getSeconds() - 10);
+
+    // just failed a call, no need to validate
+    assertThat(provider.needsToBeValidated()).isFalse();
+
+    // advance to go after
+    timeProvider.advanceTimeBySeconds(Constants.ETH1_FAILED_ENDPOINT_CHECK_INTERVAL.getSeconds());
+
+    // after the interval needs to be validated
+    assertThat(provider.needsToBeValidated()).isTrue();
+  }
+
+  private void prepareRequestWithSyncingResponse(Request request, boolean isSyncing) {
+    EthSyncing response = new EthSyncing();
+    EthSyncing.Result result = new EthSyncing.Result();
+    result.setSyncing(isSyncing);
+    response.setResult(result);
+    when(request.sendAsync()).thenReturn(SafeFuture.completedFuture(response));
+  }
+
+  private void prepareRequestWithChainidResponse(Request request, String chainId) {
+    EthChainId response = new EthChainId();
+    response.setResult(chainId);
+    when(request.sendAsync()).thenReturn(SafeFuture.completedFuture(response));
+  }
+
+  private void prepareFailingRequest(Request request) {
+    when(request.sendAsync()).thenReturn(SafeFuture.failedFuture(new RuntimeException("error")));
+  }
+
+  private void failAValidation() throws ExecutionException, InterruptedException {
+    prepareRequestWithChainidResponse(request1, CHAIN_ID_WRONG);
+    when(web3.ethChainId()).thenReturn(request1);
+    prepareRequestWithSyncingResponse(request2, false);
+    when(web3.ethSyncing()).thenReturn(request2);
+
+    assertThat(provider.validate().get()).isFalse();
+    assertThat(provider.isValid()).isFalse();
+  }
+
+  private void succeedAValidation() throws ExecutionException, InterruptedException {
+    prepareRequestWithChainidResponse(request1, CHAIN_ID_CORRECT);
+    when(web3.ethChainId()).thenReturn(request1);
+    prepareRequestWithSyncingResponse(request2, false);
+    when(web3.ethSyncing()).thenReturn(request2);
+
+    assertThat(provider.validate().get()).isTrue();
+    assertThat(provider.isValid()).isTrue();
+  }
+
+  private void failACall() {
+    prepareFailingRequest(request1);
+    when(web3.ethSyncing()).thenReturn(request1);
+
+    assertThat(provider.ethSyncing()).isCompletedExceptionally();
+  }
+
+  private void succeedACall() throws ExecutionException, InterruptedException {
+    prepareRequestWithSyncingResponse(request1, true);
+    when(web3.ethSyncing()).thenReturn(request1);
+
+    assertThat(provider.ethSyncing().get()).isTrue();
```

Comment:
```suggestion
    assertThat(provider.validate()).isCompletedWithValue(true);
```

And then you won't need to declare `throws ExcecutionException, InterruptedException` anywhere in the test either.